packages/liveblocks-core/src/room.ts

```diff
@@ -3088,11 +3097,17 @@ export function createRoom<
       reconnect: () => managedSocket.reconnect(),
       disconnect: () => managedSocket.disconnect(),
       destroy: () => {
-        syncSourceForStorage.destroy();
+        eventHub.roomWillDestroy.notify();
         context.yjsProvider?.off("status", yjsStatusDidChange);
+        syncSourceForStorage.destroy();
         syncSourceForYjs.destroy();
         uninstallBgTabSpy();
         managedSocket.destroy();
+
+        // Unregister all registered callbacks
+        for (const source of Object.values(eventHub)) {
+          source[Symbol.dispose]();
+        }
```

Comment:
I added this because I realized there was a cleanup bug here, but feel free to remove it from this PR, if you think this is causing the awareness error. I can try to tackle that in a separate PR instead of blocking this PR with it. Whatever you prefer!